File: backend/app/models/test_result.py
from dataclasses import dataclass
from datetime import datetime
from typing import Any

from app.models.status import TestStatus


@dataclass
class TestResult:
    test_run_id: str
    test_name: str
    status: TestStatus
    duration_ms: float
    timestamp: datetime
    error_message: str | None = None
    measurements: dict[str, Any] | None = None

File: frontend/src/app/features/test-runs/models/test-result.ts
import { TestStatus } from '../../../shared/models/test-status';

export interface TestResult {
  id: string;
  test_run_id: string;
  test_name: string;
  status: TestStatus;
  duration_ms: number;
  timestamp: string;
  error_message: string | null;
  measurements: Record<string, unknown> | null;
}